### PR TITLE
Surface_mesh: Add move semantics

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

WIP to add move semantics

## Release Management

* Affected package(s): Surface_mesh

### CODE REVIEW

File: Surface_mesh/include/CGAL/Surface_mesh/Properties.h [hunk 1]
```diff
@@ -494,13 +507,13 @@ class Property_container
     void swap (Property_container& other)
     {
       this->parrays_.swap (other.parrays_);
-      std::swap(this->size_, other.size_);
+      std::swap(this->size_, other.size_); // AF: why not the same for capacity_  ?
     }
```
Comment (lrineau) [inline Surface_mesh/include/CGAL/Surface_mesh/Properties.h]: I agree that is strange not to swap `capacity_` as well.

**Review by lrineau — APPROVED** [review]

**Comment by lrineau** [comment]
Do we want to risk this PR in CGAL-5.5?

**Comment by afabri** [comment]
> Do we want to risk this PR in CGAL-5.5?

How unsure are you about your review and changes?  So far it is only used in one plugin of the demo,    Do you fear that a user with a `std::vector<Surface_mesh<..>>`    might encounter problems?

**Review by mglisse — COMMENTED** [review]
@lrineau It looks sensible, but I am not familiar with this package.

File: Surface_mesh/test/Surface_mesh/surface_mesh_test.cpp [hunk 2]
```diff
@@ -230,6 +230,40 @@ void properties () {
...
+  assert(num_faces(m2) == 0);
+
+  // test copy-assignment
+  m2 = f.m;
+  assert(f.m.is_valid());
+  assert(m2.is_valid());
+  assert(num_faces(f.m) == nf);
+  assert(num_faces(m2) == nf);
+
+  // test copy-assignment
```
[the remark this answers is not in the document]
Comment (lrineau) [reply Surface_mesh/test/Surface_mesh/surface_mesh_test.cpp:259]: No, I wanted to check the copy-constructor. I will change that...

File: Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h [hunk 3]
```diff
@@ -343,6 +343,18 @@ class Surface_mesh
     typedef typename Base::reference reference;
     Property_map() : Base() {}
     Property_map(const Base& pm): Base(pm) {}
+
+    Property_map(const Property_map& pm) = default;
+
+    Property_map(Property_map&& pm) noexcept
+      : Base(std::move(pm))
+    {}
```
[the remark this answers is not in the document]
Comment (lrineau) [reply Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h:351]: You are right: that class `Property_map` could use the rule of zero.

[the remark this answers is not in the document]
Comment (lrineau) [reply Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h]: Fixed with commit https://github.com/CGAL/cgal/pull/6495/commits/3072b9394289a54b078e0a06ff1c34f85f639ed9. Please re-review.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-77

**Pull request merged**